pkg/core/transactor/commands.go

```diff
@@ -0,0 +1,225 @@
+package transactor
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+
+	"math/big"
+
+	ristretto "github.com/bwesterb/go-ristretto"
+	cfg "github.com/dusk-network/dusk-blockchain/pkg/config"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/database"
+	"github.com/dusk-network/dusk-blockchain/pkg/wallet"
+	walletdb "github.com/dusk-network/dusk-blockchain/pkg/wallet/database"
+	"github.com/dusk-network/dusk-blockchain/pkg/wallet/transactions"
+	"github.com/dusk-network/dusk-wallet/key"
+)
+
+var testnet = byte(2)
+
+func (t *Transactor) loadWallet(password string) (string, error) {
+	// First load the database
+	db, err := walletdb.New(cfg.Get().Wallet.Store)
+	if err != nil {
+		return "", err
+	}
+
+	// Then load the wallet
+	w, err := wallet.LoadFromFile(testnet, db, fetchDecoys, fetchInputs, password)
+	if err != nil {
+		db.Close()
+		return "", err
+	}
+
+	walletAddr, err := w.PublicAddress()
+	if err != nil {
+		db.Close()
+		return "", err
+	}
+
+	t.w = w
+	return walletAddr, nil
+}
+
+func (t *Transactor) createWallet(password string) (string, error) {
+	db, err := walletdb.New(cfg.Get().Wallet.Store)
+	if err != nil {
+		return "", err
+	}
+
+	w, err := wallet.New(rand.Read, testnet, db, fetchDecoys, fetchInputs, password)
+	if err != nil {
+		db.Close()
+		return "", err
+	}
+
+	walletAddr, err := w.PublicAddress()
+	if err != nil {
+		db.Close()
+		return "", err
+	}
+
+	t.w = w
+	return walletAddr, nil
+}
+
+func (t *Transactor) createFromSeed(seed string, password string) (string, error) {
+
+	seedBytes, err := hex.DecodeString(seed)
+	if err != nil {
+		return "", fmt.Errorf("error attempting to decode seed: %v\n", err)
+	}
+
+	// First load the database
+	db, err := walletdb.New(cfg.Get().Wallet.Store)
+	if err != nil {
+		return "", err
+	}
+
+	// Then load the wallet
+	w, err := wallet.LoadFromSeed(seedBytes, testnet, db, fetchDecoys, fetchInputs, password)
+	if err != nil {
+		db.Close()
+		return "", err
+	}
+
+	walletAddr, err := w.PublicAddress()
+	if err != nil {
+		db.Close()
+		return "", err
+	}
+
+	t.w = w
+	return walletAddr, nil
+}
+
+func (t *Transactor) CreateStandardTx(amount uint64, address string) (transactions.Transaction, error) {
+	if err := t.syncWallet(); err != nil {
+		return nil, err
+	}
+
+	// Create a new standard tx
+	// TODO: customizable fee
+	tx, err := t.w.NewStandardTx(cfg.MinFee)
+	if err != nil {
+		return nil, err
+	}
+
+	// Turn amount into a scalar
+	amountScalar := ristretto.Scalar{}
+	amountScalar.SetBigInt(big.NewInt(0).SetUint64(amount))
+
+	// Send amount to address
+	if err := tx.AddOutput(key.PublicAddress(address), amountScalar); err != nil {
+		return nil, err
+	}
+
+	// Sign tx
+	err = t.w.Sign(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	return tx, nil
+}
+
+func (t *Transactor) CreateStakeTx(amount, lockTime uint64) (transactions.Transaction, error) {
+	if err := t.syncWallet(); err != nil {
```

Comment:
If we are now syncing with each accepted block, can this call be removed from all tx creation methods, and the balance method?

Reply:
I decided to leave it for the situations where walletDB is moved to another node and a wallet call is sent before next acceptedBlock is accepted. To handle such corner case we could provide CLI command syncWallet. 
If you insist, we could ignore this and remove syncWallet for now.

Reply:
I'm not sure of a common scenario where we would move the walletDB out of the executable folder (in which case it can also just be copied). Thanks to the accepted block update we can always be sure we are synced to whatever our local chain tip is, so I believe this would be redundant.

Reply:
Cool. Removing it. 
In this case we might need to document that after calling loadwallet, user will need to wait for the next acceptedBlock before sending any other wallet requests (bid,stake,transfer, balance)

Reply:
The wallet will still be synced with the last block it accepted on startup. But it's a good idea to maybe mention that wallet calls may fail as the node may be syncing when it's just started.

Reply:
Note that this will break the E2E as it uses wallets with outputs in genesis block but the block at height 0 is never populated as acceptedBlock.

Reply:
Hmm, I hadn't considered it. Could we propagate the chain tip after setting up the chain?